src/dehb/optimizers/de.py

```diff
@@ -525,20 +553,20 @@ def sample_mutants(self, size, population=None):
 
         return mutants
 
-    def run(self, generations=1, verbose=False, budget=None, reset=True, **kwargs):
+    def run(self, generations=1, verbose=False, fidelity=None, reset=True, **kwargs):
         # checking if a run exists
         if not hasattr(self, 'traj') or reset:
             self.reset()
             if verbose:
                 print("Initializing and evaluating new population...")
```

Comment:
This caught my eye. 

This `traj` could be confusing as its role of `reset` appears important.

Given we have moved to ResultItem and nice data classes, can we also store these `traj` like data variables better?
If you have a solution in mind, please feel free to raise an issue with it!

Reply:
This code block seems rather unituitive for me anyways. We are restriciting the user to always start a new run when calling the `run` method. I would personally prefer leaving this check out generally and leaving the users to reset the optimizer when ever they want. This would then be similar to the behavior in `DEHB`.

For the `traj` like classes I do agree to store them in a more principled fashion. I will think about it!